src/etc/inc/filter.inc

```diff
@@ -2048,7 +2048,7 @@ function filter_generate_address(&$FilterIflist, &$rule, $target = 'source', $is
         $matches = "";
         if ($network_name == '(self)') {
             $src = "(self)";
-        } elseif (preg_match("/(.*)ip$/", $rule[$target]['network'], $matches)) {
+        } elseif (preg_match("/(^[wan|lan|opt]+[0-9]*)ip$/", $rule[$target]['network'], $matches)) {
```

Comment:
First one must be round parenthesis, no plus at the end. Lan or wan don't have numbers so shuffle number match into first as suffix to opt with a plus instead of asterisk.